processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -823,8 +826,8 @@ func (c *WatchClient) extractPodAttributes(pod *api_v1.Pod) map[string]string {
 					tags[string(conventions.ServiceNameKey)] = ref.Name
 				}
 				if c.Rules.DeploymentName || c.Rules.ServiceName {
-					if replicaset, ok := c.GetReplicaSet(string(ref.UID)); ok {
-						name := replicaset.Deployment.Name
+					if c.Rules.DeploymentNameFromReplicaSet {
```

Comment:
consider a small refactor to avoid duplication:

```golang
var deploymentName string
if c.Rules.DeploymentNameFromReplicaSet {
    deploymentName = extractDeploymentNameFromReplicaSet(ref.Name)
} else if replicaset, ok := c.GetReplicaSet(string(ref.UID)); ok {
    deploymentName = replicaset.Deployment.Name
}
if deploymentName != "" {
    if c.Rules.DeploymentName {
	    tags[string(conventions.K8SDeploymentNameKey)] = deploymentName
	}
	if c.Rules.ServiceName {
		// deployment name wins over replicaset name
		tags[string(conventions.ServiceNameKey)] = deploymentName
	}
}
```